tests/Common/SpanChecker.php

```diff
@@ -138,4 +151,25 @@ public function flattenTraces($traces)
 
         return $result;
     }
+
+    /**
+     * PHP < 5.6 does not offer a way to filter only specific elements in an array by key.
+     *
+     * @param array $associative
+     * @param string[] $allowedKeys
+     * @param bool $include
+     * @return array
+     */
+    private function filterArrayByKey54($associative, $allowedKeys, $include = true)
```

Comment:
Without reading the docblock comment, the "54" in the method name is a bit confusing (I saw the usage before the definition so that why I was confused). 😄 But also since this affects PHP 5.5 as well. And we can also make it static if we want to. :)

```suggestion
    private static function filterArrayByKey($associative, $allowedKeys, $include = true)
```

Maybe we do a check for the PHP version and use `array_filter()` with the supported constants and fall back to this code for older versions? Or maybe we just keep the method body as is and not worry about it. :)

Reply:
Good catch! Renaming an moving to static! If you don't mind I would not move to double implementation (pre and post 5.5) as I think it'd add some noise. What do you think?